atlp-rwanda/stackup2-barefoot-backend
# #171380753 create return trip user request

#### What does this PR do?
Creates API endpoint to receive return trip request from a user

#### Description of Task to be completed?
- create request model and its migration
- create createTripRequest function in request.controller.js file
- create handleTripRequest function in request.service.js file
- create  trips route **POST /api/trips** in request.routes file
- create return trip tests in request.test.js file

#### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- Make sure to rename .env.example to .env, 
- In .env file, provide all values for variables under `#Development database`  and `#General`
- Finally, run the following commands respectively:
`git fetch`
`git checkout ft-return-trip-171380753`
`npm i`
`npx sequelize-cli db:create`
`npx sequelize-cli db:migrate`
`npm run start`
- users and trips table will be created in your database
- To create a new return trip request, send the following data in your request body as JSON object to `POST http://<your_server_hostname>:<server_port>/api/trips` endpoint:
```json
    travelFrom: 'kibuye',
    travelTo: 'Gasabo',
    travelDate: '2020-03-16',
    returnDate:'2020-03-26'
    travelReason: 'vacation',
    travelType:'return-trip',
    accommodation: true,
```
#### Any background context you want to provide?
You need to log in because the obtained token will be sent in return trip route

#### What are the relevant pivotal tracker stories?
[#171380753](https://www.pivotaltracker.com/story/show/171380753)

#### Screenshots (if appropriate)
![CaptureA](https://user-images.githubusercontent.com/52660542/77148076-e4566e00-6a32-11ea-91fd-ee02a2bf50a1.PNG)
![CaptureB](https://user-images.githubusercontent.com/52660542/77148093-ed473f80-6a32-11ea-800a-a52f912b1298.PNG)
![CaptureC](https://user-images.githubusercontent.com/52660542/77148107-f2a48a00-6a32-11ea-952e-58391c3de0ac.PNG)
![CaptureD](https://user-images.githubusercontent.com/52660542/77148126-ffc17900-6a32-11ea-8d07-588f248e5a49.PNG)
![CaptureE](https://user-images.githubusercontent.com/52660542/77148143-051ec380-6a33-11ea-8915-6c5cdf5ee817.PNG)
![CaptureF](https://user-images.githubusercontent.com/52660542/77148162-0ea82b80-6a33-11ea-9d11-196d5e635070.PNG)
